lib/routes/route.js

```diff
@@ -331,11 +340,66 @@ module.exports = function(app, router) {
                 logger.error(`Could not find image of ID ${id} for direct link, serving "removed" placeholder image via redirect`);
                 res.redirect("/images/removed.png");
             } else {
-                res.status(200);
-                res.append("Cache-Control", "public, max-age=2592000");
-                res.type(imageEntry[0].mimetype);
-                res.send(imageEntry[0].data.buffer); // Send the file data to the browser.
-                logger.info("Served image " + imageEntry[0].id + " by direct link");
+                if(req.query.res && req.query.res === "low") {
+                    var extension = req.params.ext;
+                    if(extension === "jpeg") {
+                        sharp(imageEntry[0].data.buffer).jpeg({quality: 40}).toBuffer().then((imageBuffer) => {
+                            serverImage(imageBuffer);
+                            logger.info("Served low resolution image " + imageEntry[0].id + " by direct link");
+                        }).catch(() => {
+                            serverImage(imageEntry[0].data.buffer);
+                            logger.error(`Could not reduce the image to low resolution, serving original image.`);
```

Comment:
Do you think it makes sense to move this duplicated handler code into a separate method that is then used for all the other fail cases?